src/shared/settingsholder.cpp

```diff
@@ -162,24 +155,25 @@ void SettingsHolder::setRawSetting(const QString& key, const QVariant& value) {
 QString SettingsHolder::getReport() const {
   QString buff;
   QTextStream out(&buff);
-  auto settingsKeys = m_settings.childKeys();
-  for (const auto& setting : settingsKeys) {
-    if (SENSITIVE_SETTINGS.contains(setting)) {
-      out << setting << " -> <Sensitive>" << Qt::endl;
-      continue;
-    }
-    out << setting << " -> ";
-    QVariant value = m_settings.value(setting);
-    switch (value.typeId()) {
-      case QVariant::List:
-      case QVariant::StringList:
-        out << '[' << value.toStringList().join(",") << ']' << ' ';
-        break;
-      default:
-        out << value.toString();
-    }
-    out << Qt::endl;
+#define SETTING(type, toType, getter, setter, remover, has, key, defvalue, \
+                userSettings, removeWhenReset, sensitive)                  \
+  if (sensitive) {                                                         \
```

Comment:
`if (has()) {`
  ... from line 160 to the end

because not all the keys exist. Do we want to show them also if not existing for the current device?